src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs

```diff
@@ -509,15 +509,16 @@ private void PruneCache()
                 if (node.IsPersisted)
                 {
                     if (_logger.IsTrace) _logger.Trace($"Removing persisted {node} from memory.");
-                    if (node.Keccak is null)
+                    Keccak? keccak = node.Keccak;
+                    if (keccak is null)
                     {
-                        node.ResolveKey(this, true); // TODO: hack
-                        if (node.Keccak != key)
+                        keccak = node.GenerateKey(this, isRoot: true);
+                        if (keccak != key)
                         {
-                            throw new InvalidOperationException($"Persisted {node} {key} != {node.Keccak}");
+                            throw new InvalidOperationException($"Persisted {node} {key} != {keccak}");
                         }
```

Comment:
Oh that is why it didn't work! We may remove this code probably and just go with `_dirtyNodes.Remove(key);`

Reply:
Is a sensible safety check? Just shouldn't be altering the actual TrieNode by forcing a Keccak on it?

Reply:
Since it is getting called 
![image](https://github.com/NethermindEth/nethermind/assets/1142958/742f0fde-a542-4a98-9476-52739c9a55d1)